### PR TITLE
[For 0.18.x] fix: Instancing fixes

### PR DESCRIPTION
**Pull request by schinkowski**
We had an architectural problem with the way we send instances, which this PR addresses.
Previously we sent one instance info for each instance, which caused problems when the same object is instanced on multiple parents.
With these changes each instance has a collection of parents it appears on, which allows us to parent the instances to the correct parent and allows us to do nested instances correctly.

### CODE REVIEW

**Comment by sindharta-tanuwijaya** [comment]
Can you elaborate what kind of particular bug that this PR fixes ?

**Comment by schinkowski** [comment]
> Can you elaborate what kind of particular bug that this PR fixes ?



> Can you elaborate what kind of particular bug that this PR fixes ?

I updated the description.

File: Runtime/Scripts/BaseMeshSync.cs [hunk 1]
```diff
@@ -2318,7 +2364,14 @@ private static bool HandleKeywords(Material destMat, string keyword)
         Matrix4x4 localMatrix = objTransform.localToWorldMatrix;
         for (int x = 0; x < 3; x++)
         for (int y = 0; y < 3; y++)
+        {
+            if (Math.Abs(localMatrix[x, y] - converted[x, y]) > 0.01f)
```
Comment (sindharta-tanuwijaya) [inline Runtime/Scripts/BaseMeshSync.cs]: What is this for ?
Can this be separated into a different PR ?

File: Runtime/Scripts/BaseMeshSync.cs [hunk 2]
```diff
@@ -2242,48 +2307,24 @@ private static bool HandleKeywords(Material destMat, string keyword)
...
-            // Set the transform relative to `MainObject`.
-            // The easiest and safest way to do that is to:
-            // - parent the instanced copy to `MainObject`
-            // - set the transform matrix
-            // - restore the original parent
-            var originalParent = instancedCopy.parent;
+    }
 
-            instancedCopy.transform.SetParent(instanceRendererParent.transform);
+    static void SetInstanceTransformRelative(
```
Comment (sindharta-tanuwijaya) [inline Runtime/Scripts/BaseMeshSync.cs]: Where is this function called ?

Comment (schinkowski) [reply Runtime/Scripts/BaseMeshSync.cs]: We're not using it anymore, I've removed it now.

File: Runtime/Scripts/BaseMeshSync.cs (hunk 1 above)
Comment (schinkowski) [reply Runtime/Scripts/BaseMeshSync.cs]: Deleted now
